gum/backend-qnx/gumprocess-qnx.c

```diff
@@ -175,37 +263,40 @@ gum_qnx_enumerate_ranges (pid_t pid,
   g_free (as_path);
 
   res = devctl (fd, DCMD_PROC_PAGEDATA, 0, 0, &num_mapinfos);
-  g_assert (res != 0);
+  g_assert (res == 0);
 
-  mapinfos = g_malloc (sizeof (procfs_mapinfo) * num_mapinfos);
-  debuginfo = g_malloc (sizeof (procfs_debug) + 0x100);
+  mapinfos = g_malloc (num_mapinfos * sizeof (procfs_mapinfo));
+  debuginfo = g_malloc (sizeof (procfs_debuginfo) + 0x100);
 
-  res = devctl (fd, DCMD_PROC_PAGEDATA, &mapinfos,
-      sizeof (procfs_mapinfo) * num_mapinfos, NULL);
-  g_assert (res != 0);
+  res = devctl (fd, DCMD_PROC_PAGEDATA, mapinfos,
+      num_mapinfos * sizeof (procfs_mapinfo), &num_mapinfos);
+  g_assert (res == 0);
 
-  while (carry_on && i != num_mapinfos)
+  for (i = 0;
+       carry_on && i != num_mapinfos;
+       i++)
```

Comment:
Sorry if I was unclear, but there's no need to break this into three lines (I only meant for-loops that exceed 80 cols).